### PR TITLE
feat: option to hide zero-balance tokens

### PR DESCRIPTION
**Pull request by tuliomir**
This feature adds a new option in the wallet settings, “Hide zero-balance tokens”. When checked, any tokens with a balance of zero would not be displayed anywhere in the wallets.

### Acceptance Criteria
- When “Hide zero-balance tokens” is checked, any tokens without a balance should be hidden everywhere in the wallet.
- If the user unchecks the option again, they should be shown.
- User should be allowed to register zero-balance tokens even if the flag is active.
- User should be able to create exceptions for selected tokens ( "Always show" flag )
  - [ ] #310 

#### Changes demonstration
The video below shows a wallet with registered and unregistered tokens with zero balance. When the "hide" flag is activated on the settings screen, those tokens are hidden from the user on all screens.

https://user-images.githubusercontent.com/1299409/167445861-fc63d095-db50-4f5b-8310-6793fe0d0d3a.mp4

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/utils/wallet.js [hunk 1]
```diff
@@ -831,6 +873,38 @@ const wallet = {
...
+  /**
+   * Checks if the zero-balance tokens are set to be hidden
+   *
+   * @return {boolean} If the tokens are hidden
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  areZeroBalanceTokensHidden() {
+    return storage.getItem('wallet:hide_zero_balance_tokens') !== false;
```
Comment (andreabadesso) [inline src/utils/wallet.js:885]: Maybe create a constant for that attribute so we don't have to repeat it on the methods

File: src/screens/Settings.js [hunk 2]
```diff
@@ -187,6 +207,17 @@ class Settings extends React.Component {
...
             <p><strong>{t`Allow notifications:`}</strong> {this.state.isNotificationOn ? <span>{t`Yes`}</span> : <span>{t`No`}</span>} <a className='ml-3' href="true" onClick={this.toggleNotificationSettings}> {t`Change`} </a></p>
+            <p>
+              <strong>{t`Hide zero-balance tokens:`}</strong> {
+              this.state.zeroBalanceTokensHidden
+                ? <span>{t`Yes`}</span>
+                : <span>{t`No`}</span>
+              }
+              <a className="ml-3" href="true" onClick={this.toggleZeroBalanceTokens}> {t`Change`} </a>
+              <i className="fa fa-question-circle pointer ml-3"
+                 title={t`When selected, any tokens with a balance of zero will not be displayed anywhere in the wallets.`}>
```
Comment (andreabadesso) [inline src/screens/Settings.js:218]: Did you run `make update_pot` to add this new string to locale? We should also make sure that it is translated to portuguese

**Review by andreabadesso — CHANGES_REQUESTED** [review]
I like the UI, but should we also indicate that there are zero-balance tokens hidden? Or at least the ones we already interacted with (at least 1 tx)

File: src/screens/Settings.js (hunk 2 above)
Comment (tuliomir) [reply src/screens/Settings.js:218]: Done on [6cfa920](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/6cfa9204e4de33859e0dabb7892b62cb5ae244e9)

File: src/utils/wallet.js (hunk 1 above)
Comment (tuliomir) [reply src/utils/wallet.js]: Refactored these strings to a constant on [e52499f](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/e52499f06acec98c58d416b081819ae55dbfa2db).

Also centralized other storage access keys on this file.

File: src/components/TokenBar.js [hunk 3]
```diff
@@ -124,13 +124,23 @@ class TokenBar extends React.Component {
...
+    const unknownTokens = this.getUnknownTokens(shouldHideZeroBalanceTokens);
 
     const renderTokens = () => {
       return this.props.registeredTokens.map((token) => {
+        const tokenUid = token.uid;
+        const isTokenHTR = tokenUid === hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
+        const tokenBalance = this.getTokenBalance(tokenUid);
+
+        // Skip every token without balance, except HTR, if the hiding flag is active
+        if (shouldHideZeroBalanceTokens && !isTokenHTR && tokenBalance === '0.00') {
```
Comment (pedroferreira1) [inline src/components/TokenBar.js:137]: This probably won't show new registered tokens with zero balance, as described in the project doc.

**Review by pedroferreira1 — CHANGES_REQUESTED** [review]

File: src/screens/Settings.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/screens/Settings.js]: Fixed on [553bb6f](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/553bb6f6d17af2b9a88fa10d44ba1a038ccea8cb) :+1:

File: src/components/TokenBar.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/components/TokenBar.js]: It was working, but I was also uncomfortable with validating it via string.
Refactored this code to use the raw integer on [72bb4a8](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/72bb4a8c630a6966473c0309aba7a2dd91de0af5).

File: src/screens/Settings.js [hunk 4]
```diff
@@ -121,7 +125,23 @@ class Settings extends React.Component {
   }
 
   /**
-   * Called after user confirms the notification toggle action  
+   * Activates or deactivates the option to hide zero-balance tokens from the UI.
+   * @param {Event} e
+   */
+  toggleZeroBalanceTokens = (e) => {
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/screens/Settings.js:133]: Good point.
Added the confirmation modal on [003ed60](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/003ed60b88578a332e715149f34934caf17cef08) and and its new translation strings on [02b8bc1](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/02b8bc14c68763c01bc39a66004a6748bf910412)

**Review by andreabadesso — APPROVED** [review]

File: src/components/TokenBar.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply src/components/TokenBar.js]: It probably was working for custom tokens, or NFTs that were not approved by us in the NFT review but not for approved NFTs.

I don't like the way you handled this, it's not good to have a method returning different parameter types depending on a boolean parameter.

I feel we should create two methods, (i) `getTokenBalance` that returns the balance as an integer, and (ii) `getTokenBalanceFormatted` that uses the first one and returns the string.

What do you think?

[the remark this answers is not in the document]
Comment (tuliomir) [reply src/components/TokenBar.js]: Implemented on [65b7c89](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/65b7c89a829250f2cbb43e896f92dd1c0dcb6ecb) :+1:

File: src/screens/Settings.js [hunk 5]
```diff
@@ -121,7 +125,51 @@ class Settings extends React.Component {
...
+      });
+    }
+    $('#confirmModal').modal('show');
+  }
+
+  /**
+   * Activates or deactivates the option to hide zero-balance tokens from the UI.
+   * @param {Event} e
+   */
+  handleToggleZeroBalanceTokens = (e) => {
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/screens/Settings.js]: That's right: this is a leftover from a previous refactoring.
Removed on [cca6142](https://github.com/HathorNetwork/hathor-wallet/pull/309/commits/cca6142dc2e31a246462fc251afb6935ab6d477a).

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**